modules/snooker/src/main/ts/ephox/snooker/api/TableSize.ts

```diff
@@ -0,0 +1,118 @@
+import { HTMLTableCellElement, HTMLTableElement } from '@ephox/dom-globals';
+import { Arr, Fun } from '@ephox/katamari';
+import { Element, Width } from '@ephox/sugar';
+import { Warehouse } from '../model/Warehouse';
+import { BarPositions, ColInfo } from '../resize/BarPositions';
+import * as ColumnSizes from '../resize/ColumnSizes';
+import * as Sizes from '../resize/Sizes';
+import * as CellUtils from '../util/CellUtils';
+
+export interface TableSize {
+  width: () => number;
+  pixelWidth: () => number;
+  getWidths: (warehouse: Warehouse, direction: BarPositions<ColInfo>, tableSize: TableSize) => number[];
+  getCellDelta: (delta: number) => number;
+  singleColumnWidth: (w: number, delta: number) => number[];
+  minCellWidth: () => number;
+  setElementWidth: (cell: Element<HTMLTableCellElement>, amount: number) => void;
+  setTableWidth: (table: Element<HTMLTableElement>, newWidths: number[], delta: number) => void;
+  label: string;
+}
+
+const noneSize = (element: Element<HTMLTableElement>): TableSize => {
+  const getWidth = () => Width.get(element);
+  const singleColumnWidth = (w: number, delta: number) => {
+    const newNext = Math.max(CellUtils.minWidth(), w + delta);
+    return [ newNext - w ];
+  };
+
+  return {
```

Comment:
Not really too sure about what the behaviour for `getCellData()`, `singleColumnWIdth` and `minCellWidth` should be here. It doesn't really matter at the moment, as they aren't used, but I'm wondering if they all should return `0` instead (ie no change and no min-width).

Thoughts?

Reply:
All of the above are used for resizing calculations, but since none can't be used with resizing it's never called. It's more what should happen if they were to be used?

Reply:
Changed to return 0, as it's already possible to get a 0 delta if nothing changed.